assets/js/modules/analytics/components/settings/GA4SettingsControls.js

```diff
@@ -116,7 +116,10 @@ export default function GA4SettingsControls() {
 
 		setPropertyID( newPropertyID );
 		setWebDataStreamID( newWebDataStreamID );
-		setMeasurementID( matchedWebDataStream?.measurementId || '' ); // eslint-disable-line sitekit/acronym-case
+		setMeasurementID(
+			// eslint-disable-next-line sitekit/acronym-case
+			matchedWebDataStream?.webStreamData?.measurementId || ''
```

Comment:
@felixarntz in this case the `matchedWebDataStream` may not be set, but when it is that value comes from `matchWebDataStream` so it should always be a web dataStream. So the condition on `webStreamData?` shouldn't really be needed here but it is intended to be fail-safe due to the unreliability of `matchedWebDataStream`.

So this could be removed for consistency but doesn't hurt due to the fallback
```suggestion
		setMeasurementID(
			// eslint-disable-next-line sitekit/acronym-case
			matchedWebDataStream?.webStreamData.measurementId || ''
```

Reply:
I guess my point is in neither case having this hurts :) I see what you mean about the `matchedWebDataStream` potentially not being set, but like you're saying even then that second `?.` is not necessary. So I think we should consistently go with one or the other approach:
* Either never use `?.` where we don't expect the initial value to not be set
* Or be defensive and always use `?.` on "sub-objects" from 3P API

I think I prefer the first approach (so basically what you are suggesting above), since at least we would get a clear failure at this point _if_ the shape of the object was suddenly different from what we're expecting.

Reply:
@felixarntz Yeah that sounds good. Being overly defensive here would mask a critical change in the API response (if it were to change unexpectedly) so I think we should only use the conditional access when it is expected to be necessary. I'll apply my suggestion above 👍